contracts/Token.sol

```diff
@@ -41,6 +44,19 @@ contract Token is
     bytes32 public constant TRANSFERER_ROLE = keccak256("TRANSFERER_ROLE");
     /// @notice The role that has the ability to pause the token. Transferring, burning and minting will not be possible while the contract is paused.
     bytes32 public constant PAUSER_ROLE = keccak256("PAUSER_ROLE");
+
+    /**
+     * @dev This empty reserved space is put in place to allow future versions to inherit
+     * from contracts that need storage. This mimics openzeppelin's approach.
+     * For more information, see https://docs.openzeppelin.com/contracts/4.x/upgradeable#storage_gaps
+     * Its size is chosen for the first used variable (allowList) to be in storage slot 1000.
+     * This is checked in the testTokenStorageGap unit test.
+     * Whenever an inheritance is added, removed or changed, the storage location of allowList and
+     * all following variables is likely to change too, with catastrophic results. So remember to update
+     * this gap in order to keep allowList at position 1000!
+     * More info at [../docs/upgradeability.md]
+     */
+    uint256[446] private __gap;
```

Comment:
Is this really necessary?

Reply:
Yes, this is very necessary. It allows us to extend the token functionality through inheritance in the future. I tested this works as expected when merging the snapshot feature into this branch. Without the gap, inheriting from ERC20Snapshot would have messed up our storage layout. Instead, I was able to reduce the gap and live happily ever after.

Also, the gap does not cost any gas, because we never touch the storage it reserves. I just confirmed this with a gas consumption diff with and without gap:
```
Ran 1 test suites: 51 tests passed, 0 failed, 0 skipped (51 total tests)
testBurn0() (gas: 0 (0.000%)) 
testDeployerDoesNotGetRole() (gas: 0 (0.000%)) 
testExceedMintingAllowance() (gas: 0 (0.000%)) 
testFailAdmin() (gas: 0 (0.000%)) 
testFailBeforeTokenTransferRequirementsNotfulfilled() (gas: 0 (0.000%)) 
testFailSetAllowanceTo0BeforeResetting() (gas: 0 (0.000%)) 
testFailSetRequirements() (gas: 0 (0.000%)) 
testLogicContractCreation() (gas: 0 (0.000%)) 
testLoseAndGainRequirements() (gas: 0 (0.000%)) 
testSetRolePauser() (gas: 0 (0.000%)) 
testSetRoleRequirements() (gas: 0 (0.000%)) 
testFailTransferPause() (gas: -22 (-0.007%)) 
testTransferUnpaused() (gas: -22 (-0.007%)) 
testAllowList0() (gas: -22 (-0.007%)) 
testFeeSettings0() (gas: -22 (-0.008%)) 
testBeforeTokenTransferRequirements2() (gas: -22 (-0.008%)) 
testDecreaseAllowance(uint256,uint256) (gas: -4 (-0.009%)) 
testBeforeTokenTransfer() (gas: -22 (-0.013%)) 
testBurnSimple(uint256) (gas: -22 (-0.015%)) 
testBurnWithRequirements(uint256) (gas: -22 (-0.016%)) 
testSetUpMinter(uint256,uint256) (gas: -22 (-0.016%)) 
testMintOnce(uint256) (gas: -22 (-0.017%)) 
testTransferTo0(address) (gas: -22 (-0.017%)) 
testFeeSettingsNoERC165() (gas: 65 (0.023%)) 
testBeforeTokenTransferRequirementsOverfulfilled() (gas: 43 (0.023%)) 
testFailBeforeTokenTransferRequirements1() (gas: -22 (-0.023%)) 
testTransferWith0Requirements() (gas: 44 (0.025%)) 
testMintAllowerDoesNotNeedAllowance(uint256) (gas: 44 (0.034%)) 
testIncreaseAllowance(uint256,uint256) (gas: 66 (0.043%)) 
testSetRequirements(uint256) (gas: -22 (-0.047%)) 
testFailZeroAllowanceMint(uint256) (gas: -22 (-0.049%)) 
testSetRoleDefaultAdmin() (gas: 23 (0.054%)) 
testFailMintAllowanceUsed(uint256) (gas: -22 (-0.054%)) 
testFailSetRequirementsAdmin() (gas: -22 (-0.054%)) 
testFailSetRequirementsWrongRole() (gas: -22 (-0.054%)) 
testSetUp() (gas: -22 (-0.076%)) 
testSetRoleTransferer() (gas: -22 (-0.079%)) 
testFailBurn0() (gas: 44 (0.081%)) 
testFailAdmin2() (gas: -22 (-0.100%)) 
testFailSetRequirementsX(address) (gas: 44 (0.106%)) 
testSetRoleMintallower() (gas: -22 (-0.107%)) 
testSetRoleBurner() (gas: -22 (-0.107%)) 
testMint0() (gas: 64 (0.108%)) 
testAcceptFeeSettings0() (gas: -22 (-0.131%)) 
testAdmin() (gas: -22 (-0.151%)) 
testIsNotAdmin(address) (gas: 23 (0.157%)) 
testTransferFrom0(address) (gas: 23 (0.190%)) 
testMinterAdmin() (gas: -22 (-0.191%)) 
testDecimals() (gas: 23 (0.278%)) 
testMintAllower(address) (gas: 44 (0.301%)) 
testMintAgain(uint256,uint256) (gas: -25643 (-3.982%)) 
Overall gas change: -25647 (-0.253%)
```

Reply:
ok